packages/ai/core/generate-image/generate-image.ts

```diff
@@ -135,7 +142,9 @@ Only applicable for HTTP-based providers.
           size,
           aspectRatio,
           seed,
-          providerOptions: providerOptions ?? {},
+          providerOptions: providerName
+            ? { [providerName]: generateProviderOptions }
+            : {},
```

Comment:
Is there a use case where providerOptions has more than one key? If so this change would break it. We don't pass through provider options because we want to remove the `maxImagesPerCall` option first, otherwise it ends up in the API request body.